runtime/cudaq/platform/default/rest/helpers/infleqtion/InfleqtionServerHelper.cpp

```diff
@@ -0,0 +1,315 @@
+  static constexpr const char *DEFAULT_VERSION = "v0.2.0";
+
+public:
+  /// @brief Returns the name of the server helper.
+  const std::string name() const override { return "infleqtion"; }
+
+  /// @brief Returns the headers for the server requests.
+  RestHeaders getHeaders() override;
+
+  /// @brief Initializes the server helper with the provided backend
+  /// configuration.
+  void initialize(BackendConfig config) override;
+
+  /// @brief Creates a quantum computation job using the provided kernel
+  /// executions and returns the corresponding payload.
+  ServerJobPayload
+  createJob(std::vector<KernelExecution> &circuitCodes) override;
+
+  /// @brief Extracts the job ID from the server's response to a job submission.
+  std::string extractJobId(ServerMessage &postResponse) override;
+
+  /// @brief Constructs the URL for retrieving a job based on the server's
+  /// response to a job submission.
+  std::string constructGetJobPath(ServerMessage &postResponse) override;
+
+  /// @brief Constructs the URL for retrieving a job based on a job ID.
+  std::string constructGetJobPath(std::string &jobId) override;
+
+  /// @brief Checks if a job is done based on the server's response to a job
+  /// retrieval request.
+  bool jobIsDone(ServerMessage &getJobResponse) override;
+
+  /// @brief Processes the server's response to a job retrieval request and
+  /// maps the results back to sample results.
+  cudaq::sample_result processResults(ServerMessage &getJobResponse,
+                                      std::string &jobId) override;
+
+private:
+  /// @brief RestClient used for HTTP requests.
+  RestClient client;
+
+  /// @brief Helper method to retrieve the value of an environment variable.
+  std::string getEnvVar(const std::string &key, const std::string &defaultVal,
+                        const bool isRequired) const;
+
+  /// @brief Helper function to get value from config or return a default value.
+  std::string getValueOrDefault(const BackendConfig &config,
+                                const std::string &key,
+                                const std::string &defaultValue) const;
+
+  /// @brief Helper method to check if a key exists in the configuration.
+  bool keyExists(const std::string &key) const;
+};
+
+// Initialize the Infleqtion server helper with a given backend configuration
+void InfleqtionServerHelper::initialize(BackendConfig config) {
+  cudaq::info("Initializing Infleqtion Backend.");
+
+  // Move the passed config into the member variable backendConfig
+  backendConfig = config;
+
+  // Set default URL and version if not provided
+  backendConfig["url"] = getValueOrDefault(config, "url", DEFAULT_URL);
+  backendConfig["version"] =
+      getValueOrDefault(config, "version", DEFAULT_VERSION);
+  backendConfig["user_agent"] = getValueOrDefault(
+      config, "user_agent", "cudaq/" + std::string(cudaq::getVersion()));
+  backendConfig["target"] =
```

Comment:
Q on using `target` keyword: I wonder if it would be cleaner to use a field / key like `qpu`, `machine`, or `device` so as to avoid confusion with CUDA-Q `target`?
Currently Python code would look like - 
```
cudaq.set_target("infleqtion", target="cq_sqorpius_simulator")
```

Reply:
The motivation for using `target=` was that it mirrored the format expected by our API. But I did want to consult on the naming convention to best maintain parity with the other CUDA-Q hardware implementations, and I think going with `machine` seems to be the most appropriate for us -- updated now.